### PR TITLE
sql: Add error doc

### PR DESCRIPTION
**Pull request by shenli**
错误码解释，以及常用错误诊断文档链接

### CODE REVIEW

File: sql/error.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+| 错误码 | 说明 |
+| --------------------- | -------------------------------------------------- |
+| 9001 | 请求 PD 超时，请检查 PD Server 状态/监控/日志以及 TiDB Server 与 PD Server 之间的网络 |
+| 9002 | 请求 TiKV 超时，请检查 TiKV Server 状态/监控/日志以及 TiDB Server 与 TiKV Server 之间的网络 |
+| 9003 | TiKV 操作繁忙，一般出现在数据库负载比较高时，请检查 TiKV Server 状态/监控/日志 |
+| 9004 | 当数据库上承载的业务存在大量的事务冲突时，会遇到这种错误，请检查业务代码 |
+| 9005 | 某个 Raft Group 不可用，如副本数目不足，出现在 TiKV 比较繁忙或者是 TiKV 节点停机的时候，请检查 TiKV Server 状态/监控/日志|
+| 9500 | 单个事务过大，原因及解决方法请参考[这里](https://github.com/pingcap/docs-cn/blob/master/FAQ.md#出现-transaction-too-large-报错怎么办)|
+| | |
```
Comment (sunhao2017) [inline sql/error.md]: Redundant tables

File: sql/README.md [hunk 2]
```diff
@@ -71,4 +71,4 @@ Table of Contents
     - [TiDB SQL 语法图](https://pingcap.github.io/sqlgram/)
 - [文档型数据库](json-functions-generated-column.md)
 + 连接器和 API
-+ 故障诊断
++ [错误码，故障诊断](error.md)
```
Comment (sunhao2017) [inline sql/README.md]: 错误码，故障诊断 => 错误码与故障诊断 ?

**Comment by shenli** [comment]
@sunhao2017 PTAL

File: sql/error.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+TiDB 兼容 MySQL 的错误码，在大多数情况下，返回和 MySQL 一样的错误码。另外还有一些特有的错误码：
+
+| 错误码 | 说明 |
+| --------------------- | -------------------------------------------------- |
+| 9001 | 请求 PD 超时，请检查 PD Server 状态/监控/日志以及 TiDB Server 与 PD Server 之间的网络 |
+| 9002 | 请求 TiKV 超时，请检查 TiKV Server 状态/监控/日志以及 TiDB Server 与 TiKV Server 之间的网络 |
+| 9003 | TiKV 操作繁忙，一般出现在数据库负载比较高时，请检查 TiKV Server 状态/监控/日志 |
+| 9004 | 当数据库上承载的业务存在大量的事务冲突时，会遇到这种错误，请检查业务代码 |
+| 9005 | 某个 Raft Group 不可用，如副本数目不足，出现在 TiKV 比较繁忙或者是 TiKV 节点停机的时候，请检查 TiKV Server 状态/监控/日志 |
+| 9500 | 单个事务过大，原因及解决方法请参考[这里](https://github.com/pingcap/docs-cn/blob/master/FAQ.md#出现-transaction-too-large-报错怎么办) |
```
Comment (queenypingcap) [inline sql/error.md]: Please use relative file path. These documents will be transferred from .md format to HTML on the website. Using the absolute file path will lead to errors.

File: sql/error.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+
+## 故障诊断
+
+参见[故障诊断文档](https://github.com/pingcap/docs-cn/blob/master/trouble-shooting.md)以及 [FAQ] (https://github.com/pingcap/docs-cn/blob/master/FAQ.md)。
```
Comment (queenypingcap) [inline sql/error.md]: ditto

**Comment by queenypingcap** [comment]
@shenli PTAL

**Comment by shenli** [comment]
@queenypingcap PTAL

File: sql/error.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+| 9002 | 请求 TiKV 超时，请检查 TiKV Server 状态/监控/日志以及 TiDB Server 与 TiKV Server 之间的网络 |
+| 9003 | TiKV 操作繁忙，一般出现在数据库负载比较高时，请检查 TiKV Server 状态/监控/日志 |
+| 9004 | 当数据库上承载的业务存在大量的事务冲突时，会遇到这种错误，请检查业务代码 |
+| 9005 | 某个 Raft Group 不可用，如副本数目不足，出现在 TiKV 比较繁忙或者是 TiKV 节点停机的时候，请检查 TiKV Server 状态/监控/日志 |
+| 9500 | 单个事务过大，原因及解决方法请参考[这里](../FAQ.md#出现-transaction-too-large-报错怎么办) |
+
+
+## 故障诊断
+
+参见[故障诊断文档](../trouble-shooting.md)以及 [FAQ] (../FAQ.md)。
```
Comment (sunhao2017) [inline sql/error.md]: [FAQ] (../FAQ.md) =>[FAQ](../FAQ.md)

**Comment by shenli** [comment]
@sunhao2017 PTAL

**Pull request merged**